cpp/example_code/s3/gtests/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+﻿# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Set the minimum required version of CMake for this project.
+cmake_minimum_required(VERSION 3.14)
+
+# Set this project's name.
+project("s3-examples-gtests")
+
+# Set the C++ standard to use to build this target.
+set(CMAKE_CXX_STANDARD 14)
+
+include(FetchContent)
+FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY https://github.com/google/googletest.git
+        GIT_TAG release-1.12.1
+)
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+#Set the location of where Windows can find the installed libraries of the SDK.
+if(MSVC)
+    string(REPLACE ";" "/aws-cpp-sdk-all;" SYSTEM_MODULE_PATH "${CMAKE_SYSTEM_PREFIX_PATH}/aws-cpp-sdk-all")
+    list(APPEND CMAKE_PREFIX_PATH ${SYSTEM_MODULE_PATH})
+endif()
+
+# Build shared libraries by default.
+if(NOT BUILD_SHARED_LIBS)
+    set(BUILD_SHARED_LIBS ON)
+endif()
+
+# Find the AWS SDK for C++ package.
+find_package(AWSSDK REQUIRED COMPONENTS s3 sts)
+
+# If the compiler is some version of Microsoft Visual C++, or another compiler simulating C++,
+# and building as shared libraries, then dynamically link to those shared libraries.
+if(MSVC AND BUILD_SHARED_LIBS)
+    add_definitions(-DUSE_IMPORT_EXPORT)
+    # Copy relevant AWS SDK for C++ libraries into the current binary directory for running and debugging.
+    list(APPEND SERVICE_LIST s3 sts)
+    #For IDE's like Xcode and Visual Studio this line will be ignored because Release/Debug 
+    #  is switched internally, but this is necessary for non-IDE builds.
```

Comment:
`    # For IDEs like Xcode and Visual Studio, this line is ignored. This is because Release/Debug 
    #  is switched internally, but this is necessary for non-IDE builds.`